docs/tutorial/go.rst

```diff
@@ -0,0 +1,349 @@
+    :dedent: 2
+
+
+Build the Go application so it can be run:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:go-build]
+    :end-before: [docs:go-build-end]
+    :dedent: 2
+
+A new binary named ``go-hello-world`` should be in the current directory.
+Run the Go application using ``./go-hello-world`` to verify that
+it works.
+
+Test the Go application by using ``curl`` to send a request to the root
+endpoint. You may need a new terminal for this, if you are using Multipass use
+``multipass shell rock-dev`` to get another terminal:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:curl-go]
+    :end-before: [docs:curl-go-end]
+    :dedent: 2
+
+The Go application should respond with ``Hello World``.
+
+The Go application looks good, so stop it for now using :kbd:`Ctrl` + :kbd:`C`.
+
+Pack the Go application into a rock
+===================================
+
+
+First, we'll need a ``rockcraft.yaml`` file. Rockcraft will automate its
+creation and tailoring for a Go application by using the
+``go`` profile:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:create-rockcraft-yaml]
+    :end-before: [docs:create-rockcraft-yaml-end]
+    :dedent: 2
+
+The ``rockcraft.yaml`` file will automatically be created in your working
+directory. Open it in a text editor and check that the ``name`` is
+``go-hello-world``. Ensure that ``platforms`` includes the architecture of
+your host. For example, if your host uses the ARM
+architecture, include ``arm64`` in ``platforms``.
+
+.. note::
+    For this tutorial, we'll use the ``name`` ``go-hello-world`` and assume
+    you are running on an ``amd64`` platform. Check the architecture of your
+    system using ``dpkg --print-architecture``.
+
+    The ``name``, ``version`` and ``platform`` all influence the name of the
+    generated ``.rock`` file.
+
+Pack the rock:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:pack]
+    :end-before: [docs:pack-end]
+    :dedent: 2
+
+.. note::
+
+    Depending on your network, this step can take a couple of minutes to finish.
+
+    ``ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS`` is required whilst the Go
+    extension is experimental.
+
+Once Rockcraft has finished packing the Go rock, you'll find a new file in
+your working directory (an `OCI <OCI_image_spec_>`_ archive) with the ``.rock``
+extension:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:ls-rock]
+    :end-before: [docs:ls-rock-end]
+    :dedent: 2
+
+.. note::
+    If you changed the ``name`` or ``version`` in ``rockcraft.yaml`` or are not
+    on an ``amd64`` platform, the name of the ``.rock`` file will be different
+    for you.
+
+    By default, the ``go-framework`` will use the ``bare`` base. You can improve
+    the developer experience changing the base to ``ubuntu@24.04``, but the
```

Comment:
Could we caveat what we mean by this? E.g., for debugging it might be better to use but for production we recommend bare?